src/store/list/reducer.ts

```diff
@@ -0,0 +1,66 @@
+import { FeedState, FeedListType } from "./types";
+import { Reducer } from "redux";
+import { FetchListActionTypes } from "./action.types";
+import { FETCH_LIST } from "./constants";
+
+const initialFeedState: FeedState = {
+  latest: {
+    total: 0,
+    next: "",
+    posts: [],
+    isLoading: false,
+  },
+  frequent: {
+    total: 0,
+    next: "",
+    posts: [],
+    isLoading: false,
+  },
+};
+
+// TODO: Reducer code should be simplified with lodash + immer
+export const feedReducer: Reducer<FeedState, FetchListActionTypes> = (
+  state = initialFeedState,
+  action
+): FeedState => {
+  switch (action.type) {
+    case FETCH_LIST.LOADING:
```

Comment:
Is it necessary to make the returned state through this? Can't we have a second switch with cases of action.listType and then return different states:

For example:
```js
switch (action.listType) {
    case FeedListType.frequent: return {...state, frequent: {...state.frequent, isLoading: true}}
    //etc
}
```
As the action would only have one FeedListType at one time.

Reply:
Hmm, we can do this, but I personally find having a nested switch case too verbose. Furthermore, as I have added in the TODO above, this would become a lot cleaner if we incorporate `lodash` + `immer`. I want to keep it a separate PR though because I haven't used them before so would like to try it out proper and add it later.